packages/DataGrid/src/components/RowIndicator/RowIndicator.js

```diff
@@ -1,79 +1,64 @@
 import React from "react";
 import PropTypes from "prop-types";
-import CheckBox from "@paprika/checkbox";
 import * as sc from "./RowIndicator.styles";
 
-const _UNCHECKED_ = "unchecked";
-
 const propTypes = {
-  hasIndexIndicator: PropTypes.bool,
   hasNumber: PropTypes.bool,
   isActiveCell: PropTypes.bool,
   isActiveRow: PropTypes.bool,
-  isChecked: PropTypes.func,
   onCheck: PropTypes.func,
   row: PropTypes.shape({}),
   rowIndex: PropTypes.number,
+  isChecked: PropTypes.bool,
+  isHeaderChecker: PropTypes.bool,
 };
 
 const defaultProps = {
-  hasIndexIndicator: true,
   hasNumber: false,
   isActiveCell: false,
   isActiveRow: false,
   row: null,
   rowIndex: null,
   onCheck: () => {},
-  isChecked: () => {
-    return _UNCHECKED_;
-  },
+  isChecked: false,
+  isHeaderChecker: false,
 };
 
-function isValueNotUnchecked(value) {
-  return value !== _UNCHECKED_;
-}
-
-export default function RowIndicator(props) {
-  const [isActive, setIsActive] = React.useState(false);
-  const { rowIndex, isChecked, hasIndexIndicator, onCheck, isActiveCell, isActiveRow, hasNumber } = props;
-
-  function handleCheck() {
-    onCheck({ rowIndex });
-  }
+function RowIndicator(props) {
+  const { rowIndex, isChecked, onCheck, isHeaderChecker, isActiveCell, isActiveRow, hasNumber } = props;
 
-  function handleMouseOver() {
-    setIsActive(true);
-  }
-
-  function handleMouseLeave() {
-    setIsActive(false);
+  function handleCheck(event) {
+    const isHeader = isHeaderChecker;
+    onCheck({ rowIndex, isHeader });
+    event.stopPropagation();
   }
 
   const indicator = hasNumber ? (
     <sc.RowIndexText hasFourDigitsOrMore={rowIndex > 999}>{rowIndex}</sc.RowIndexText>
-  ) : (
-    <sc.RowIndexText>
-      <sc.Void />
-    </sc.RowIndexText>
-  );
+  ) : null;
 
   return (
-    <sc.RowContainer
-      onMouseOver={handleMouseOver}
-      onFocus={handleMouseOver}
-      onBlur={handleMouseLeave}
-      onMouseLeave={handleMouseLeave}
-    >
-      {isActive || isActiveCell || isActiveRow || isValueNotUnchecked(isChecked({ rowIndex })) || !hasIndexIndicator ? (
+    <sc.RowContainer>
+      {hasNumber && (isActiveCell || isActiveRow) ? (
+        indicator
+      ) : (
         <sc.Checkbox>
-          <CheckBox tabIndex="-1" size="small" checkedState={isChecked({ rowIndex })} onChange={handleCheck} />
+          <input type="checkbox" checked={isChecked} onChange={() => {}} onMouseDown={handleCheck} />
```

Comment:
So this is an interesting one, our checkbox requires having an onChange method, but for this case, it's irrelevant since the user can check the checkbox either, with  `enter, space or focusing on the input and click on it directly. 

Also, the state is controlled by the consumer so is not really a need to know the state of the input only setting if its checked or not. 

Later I can try to replace it with our checkbox, but not part of this PR